pkg/microservice/aslan/core/multicluster/service/clusters.go

```diff
@@ -32,47 +37,87 @@ import (
 	e "github.com/koderover/zadig/pkg/tool/errors"
 )
 
+var namePattern = regexp.MustCompile(`^[0-9a-zA-Z_.-]{1,32}$`)
+
 type K8SCluster struct {
```

Comment:
define a ClusterCreationSpec for creating cluster. This struct is used to return a cluster object, don't add fields which you don't want to expose